catch/unit/texture/hipBindTextureToMipmappedArray.cc

```diff
@@ -0,0 +1,171 @@
+  memset(hData, 0, size);
+  for (i = 0; i < height; i++) {
+    for (j = 0; j < width; j++) {
+      hData[i * width + j] = i * width + j;
+    }
+  }
+
+  // Allocate memory for Mipmapped array and set data to mipmap_level
+  hipChannelFormatDesc channelDesc = hipCreateChannelDesc<float>();
+
+  hipMipmappedArray* mip_array_ptr;
+  HIP_CHECK(hipMallocMipmappedArray(&mip_array_ptr, &channelDesc,
+                                    make_hipExtent(orig_width, orig_height, 0), 2 * mipmap_level,
+                                    hipArrayDefault));
+
+  hipArray* hipArray = nullptr;
+  HIP_CHECK(hipGetMipmappedArrayLevel(&hipArray, mip_array_ptr, mipmap_level));
+  HIP_CHECK(hipMemcpy2DToArray(hipArray, 0, 0, hData, width * sizeof(float), width * sizeof(float),
+                               height, hipMemcpyHostToDevice));
+
+  // Set texture parameters
+  texRef.addressMode[0] = hipAddressModeWrap;
+  texRef.addressMode[1] = hipAddressModeWrap;
+  texRef.filterMode = hipFilterModePoint;
+  texRef.normalized = 0;
+
+  // Bind the array to the texture
+  HIP_CHECK(hipBindTextureToMipmappedArray(&texRef, mip_array_ptr, &channelDesc));
+
+  // Allocate device memory for result
+  float* dData = nullptr;
+  HIP_CHECK(hipMalloc(&dData, size));
+  REQUIRE(dData != nullptr);
+
+  dim3 dimBlock(16, 16, 1);
+  dim3 dimGrid(width / dimBlock.x, height / dimBlock.y, 1);
+
+  hipLaunchKernelGGL(tex2DKernel, dim3(dimGrid), dim3(dimBlock), 0, 0, dData, width, mipmap_level);
+  HIP_CHECK(hipGetLastError());
+  HIP_CHECK(hipDeviceSynchronize());
+
+  // Allocate memory on host and copy result from device to host
+  float* hOutputData = reinterpret_cast<float*>(malloc(size));
+  REQUIRE(hOutputData != nullptr);
+  memset(hOutputData, 0, size);
+  HIP_CHECK(hipMemcpy(hOutputData, dData, size, hipMemcpyDeviceToHost));
+
+  for (i = 0; i < height; i++) {
+    for (j = 0; j < width; j++) {
+      if (hData[i * width + j] != hOutputData[i * width + j]) {
+        INFO("Difference found at [ " << i << j << " ]: " << hData[i * width + j]
+                                      << hOutputData[i * width + j]);
+        REQUIRE(false);
+      }
+    }
+  }
+  HIP_CHECK(hipUnbindTexture(texRef));
+  HIP_CHECK(hipFree(dData));
+  HIP_CHECK(hipFreeArray(hipArray));
+  HIP_CHECK(hipFreeMipmappedArray(mip_array_ptr));
+  free(hData);
+}
+#endif
+
+TEST_CASE("Unit_hipTextureMipmapRef2D_Positive_Check") {
+  CHECK_IMAGE_SUPPORT
+  // Height Width Vector
+  std::vector<unsigned int> hw_vec = {2048, 1024, 512, 256, 64};
+  std::vector<unsigned int> mip_vec = {8, 4, 2, 1};
+#ifdef _WIN32
+  for (auto& hw : hw_vec) {
+    for (auto& mip : mip_vec) {
+      if ((hw / static_cast<int>(pow(2, (mip * 2)))) > 0) {
+        runMipMapTest(hw, hw, mip);
+      }
+    }
+  }
+#else
+  SUCCEED("Mipmaps are Supported only on windows, skipping the test.");
+#endif
+}
+
+TEST_CASE("Unit_hipTextureMipmapRef2D_Negative_Parameters") {
+  CHECK_IMAGE_SUPPORT
+
+#ifdef _WIN32
```

Comment:
why this?

Reply:
@mangupta @chrispaquot This has been added in line with all other tests that use API that is only functional on Windows OS. This is the case for Mipmapped Array API

Reply:
Then it's better to put the test in the json file for Linux to avoid running it on Linux.

Reply:
@chrispaquot @mangupta I have just checked, and on our system it does not build on Linux. I checked on Nvidia system. Mipmapped array APIs are not recognized

Reply:
That's odd. The API hipBindTextureToMipmappedArray is located in the map file so should be accessible on Linux.

Reply:
@mangupta Any input here?

Reply:
@rakesroy It seems all changes made in this commit have been already merged as part of https://github.com/ROCm/hip-tests/pull/92 . When you verify this, the PR can be closed.